### ISSUE TITLE
prefer-const false positive

### ISSUE THREAD

**Issue by vsemozhetbyt** [eslint/eslint#7834]
* **ESLint Version:** 3.12.2
* **Node Version:** 7.3.0
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
'prefer-const': ['error', { destructuring: 'any', ignoreReadBeforeAssign: true }]
```

**What did you do? Please include the actual source code causing the issue.**
```js
/******************************************************************************/
'use strict';
/******************************************************************************/
function test() {
  let a;
  a += '*';
  return a;
}

test();
```

**What did you expect to happen?**
No error report.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  6:3  error  'a' is never reassigned. Use 'const' instead  prefer-const

✖ 1 problem (1 error, 0 warnings)
```

**Comment by nzakas** [comment]
This looks like the correct behavior to me. The `a` can declaration can be rewritten to a `const`. Can you explain why you think this is a bug?

Cc @mysticatea

**Comment by vsemozhetbyt** [comment]
@nzakas Maybe this is an edge case. I understand this can be rewritten as `const a = 'undefined*'`, but I thought that formally the value of `a` changes there. I can't imagine any useful case for such a code (I've run into something like it in an erroneous script). So maybe you are right: if `undefined` is not really a value, so  `a` does not change there. Should I close the issue?

**Comment by nzakas** [comment]
Technically the value changes from undefined to a string, but the assignment is right after the declaration, and the value never changes after that for the lifecycle of the variable. That's why the rule is flagging it: a variable that starts out uninitialized and then has just one assignment can be replaced with s constant. 

If you're satisfied with that explanation, then yes, this issue can be closed.
